src/Analyzers/CSharp/Analyzers/ConvertTypeofToNameof/CSharpConvertTypeOfToNameOfDiagnosticAnalyzer.cs

```diff
@@ -33,7 +34,9 @@ protected override bool IsValidTypeofAction(OperationAnalysisContext context)
             // Make sure that the syntax that we're looking at is actually a typeof expression and that
             // the parent syntax is a member access expression otherwise the syntax is not the kind of
             // expression that we want to analyze
-            return node is TypeOfExpressionSyntax { Parent: MemberAccessExpressionSyntax _ };
+            return node is TypeOfExpressionSyntax { Parent: MemberAccessExpressionSyntax } &&
+                // nameof(System.Void) isn't allowed in C#.
+                ((ITypeOfOperation)context.Operation).TypeOperand.SpecialType != SpecialType.System_Void;
```

Comment:
also, if we ahve `typeof(Dictionary<,>.KeyValuePair)` we shouldn't do `nameof` here either most likely.

Reply:
i think this can be simpler.  System.Void literally cannot be referenced no matter what.  so the ony way to get this is `typeof(void)`.  so we can just do `not is TypeOfExpressionSyntax { type: not PredefinedTypeSyntax { Keyword ...`

but it's up to you.

Reply:
> also, if we ahve typeof(Dictionary<,>.KeyValuePair) we shouldn't do nameof here either most likely.

Generics are already handled here, there are existing tests for it too:

https://github.com/dotnet/roslyn/blob/d4e8ed5bc68dd4d3bb374ee73fdaf5d80ab6e2e3/src/Analyzers/Core/Analyzers/ConvertTypeofToNameof/AbstractConvertTypeOfToNameOfDiagnosticAnalyzer.cs#L87

---

> i think this can be simpler. System.Void literally cannot be referenced no matter what. so the ony way to get this is typeof(void). so we can just do not is TypeOfExpressionSyntax { type: not PredefinedTypeSyntax { Keyword ...

I'm not a big fan of using syntax when a semantic check can be done, but I went for it anyway.